### PR TITLE
deps(preference_navigation): upgrade to flutter 3.10.5 and dart 3

### PR DESCRIPTION
**Pull request by elianortega**
- update missing example to dart 3 and flutter 3
- delete old CI/CD

### CODE REVIEW

File: examples/preference_navigation/lib/home/view/home_page.dart [hunk 1]
```diff
@@ -65,12 +65,12 @@ class HomeView extends StatelessWidget {
...
   /// {@macro preferences_list}
   const PreferencesList({
-    super.key,
     required this.preferences,
+    super.key,
   });
 
   /// Current preferences stored on device.
-  final Map preferences;
+  final Map<dynamic, dynamic> preferences;
```
Comment (marcossevilla) [inline examples/preference_navigation/lib/home/view/home_page.dart:73]:
```suggestion
  final Map<String, dynamic> preferences;
```
should this be `<String, dynamic>` based on the key-value approach?

File: examples/preference_navigation/lib/home/view/home_page.dart [hunk 2]
```diff
@@ -11,7 +11,7 @@ class HomePage extends StatelessWidget {
   const HomePage({super.key});
 
   /// Returns a [MaterialPageRoute] to navigate to `this` widget.
-  static Route go() {
+  static Route<void> go() {
```
Comment (marcossevilla) [inline examples/preference_navigation/lib/home/view/home_page.dart:14]:
```suggestion
  static Route<void> route() {
```
nit

File: examples/preference_navigation/lib/preferences/view/preferences_page.dart [hunk 3]
```diff
@@ -12,7 +12,7 @@ class PreferencesPage extends StatelessWidget {
   const PreferencesPage({super.key});
 
   /// Returns a [MaterialPageRoute] to navigate to `this` widget.
-  static Route go() {
+  static Route<void> go() {
```
Comment (marcossevilla) [inline examples/preference_navigation/lib/preferences/view/preferences_page.dart:15]:
```suggestion
  static Route<void> route() {
```

File: examples/preference_navigation/lib/start/view/start_page.dart [hunk 4]
```diff
@@ -11,7 +11,7 @@ class StartPage extends StatelessWidget {
   const StartPage({super.key});
 
   /// Returns a [MaterialPageRoute] to navigate to `this` widget.
-  static Route go() {
+  static Route<void> go() {
```
Comment (marcossevilla) [inline examples/preference_navigation/lib/start/view/start_page.dart:14]:
```suggestion
  static Route<void> route() {
```

**Review by marcossevilla — CHANGES_REQUESTED** [review]

File: examples/preference_navigation/lib/home/view/home_page.dart (hunk 1 above)
Comment (elianortega) [reply examples/preference_navigation/lib/home/view/home_page.dart]: Yes it should.

**Review by marcossevilla — APPROVED** [review]

**Pull request merged**